Function call 'helper_foo' got an unexpected extra argument with type i32
